capreolus/trainer/tensorflow.py

```diff
@@ -427,34 +427,33 @@ def convert_to_tf_dev_record(self, reranker, dataset):
         tf_features = []
         tf_record_filenames = []
 
+        tf_file_id = 0
         for sample in dataset:
             tf_features.extend(reranker.extractor.create_tf_dev_feature(sample))
             if len(tf_features) > 20000:
-                tf_record_filenames.append(self.write_tf_record_to_file(dir_name, tf_features))
+                tf_record_filenames.append(self.write_tf_record_to_file(dir_name, tf_features, file_name=str(tf_file_id)))
                 tf_features = []
+                tf_file_id += 1
 
         # TPU's require drop_remainder = True. But we cannot drop things from validation dataset
         # As a workaroud, we pad the dataset with the last sample until it reaches the batch size.
         if len(tf_features) % evalbatch:
             num_elements_to_add = evalbatch - (len(tf_features) % evalbatch)
             logger.debug("Number of elements to add in the last batch: {}".format(num_elements_to_add))
             element_to_copy = tf_features[-1]
-            for i in range(num_elements_to_add):
+            for i in range(self.config["evalbatch"]):
                 tf_features.append(copy(element_to_copy))
```

Comment:
@crystina-z is this correct?

Reply:
I just realized maybe here we should always append regardless of how many samples we have in `tf_features` (unless we keep tracking the *total* number of dev samples we wrote to tfrecord).